teslajsonpy/homeassistant/homelink.py

```diff
@@ -0,0 +1,90 @@
+#  SPDX-License-Identifier: Apache-2.0
+"""
+Python Package for controlling Tesla API.
+
+For more details about this api, please refer to the documentation at
+https://github.com/zabuldon/teslajsonpy
+"""
+
+from teslajsonpy.exceptions import HomelinkError
+from teslajsonpy.homeassistant.vehicle import VehicleDevice
+
+
+class TriggerHomelink(VehicleDevice):
+    """Home-Assistant class for trigger homelink of Tesla vehicles."""
+
+    def __init__(self, data, controller):
+        """Initialize the trigger homelink for the vehicle.
+
+        Parameters
+        ----------
+        data : dict
+            The trigger homelink for a Tesla vehicle.
+            https://tesla-api.timdorr.com/vehicle/commands/homelink
+        controller : teslajsonpy.Controller
+            The controller that controls updates to the Tesla API.
+
+        Returns
+        -------
+        None
+
+        """
+        super().__init__(data, controller)
+        self.type = "trigger homelink"
+        self.hass_type = "button"
+        self.name = self._name()
+        self.uniq_name = self._uniq_name()
+        self.__longitude = None
+        self.__latitude = None
+        self.__homelink_device_count = None
+        self.__homelink_nearby = None
+        self.__homelink_available = False
+
+    async def async_update(self, wake_if_asleep=False, force=False):
```

Comment:
I'm not sure if HA requires it for buttons but if it doesn't, perhaps we don't need to have an update since we don't have state. It doesn't hurt to have it though I guess.

Reply:
hmm, I'm not sure either. I added it as it was also implemented in alerts.py for the other buttons.
Regardless, wouldn't it be needed to call it's refresh() impl to get homelink availability and location, etc?

Reply:
I guess since you're storing data, then we do need this. Yes, you'd have to do a refresh to get an instant update. Otherwise there will be a lag between the current state.

Reply:
I added an update (forced) at the start of trigger_homelink to ensure we have fresh location and homelink info before trying to send the api call. this takes care of the edge case mentioned below where you might have just returned home and attempt to trigger homelink before the location is updated.

Reply:
Ok.